lib/backend/src/Obelisk/Backend.hs

```diff
@@ -277,29 +279,73 @@ renderGhcjsFrontend urlEnc ghcjsWidgets route configs f = do
+wasmScriptPreload allJsUrl (jsaddleAssets, wasmUrl) = do
+  let sendMsgWorkerJs = jsaddleAssets <> "/jsaddle_sendMsgWorker.js"
+      jsaddleJs = jsaddleAssets <> "/jsaddle.js"
+      workerRunnerJs = jsaddleAssets <> "/worker_runner.js"
+
+  elAttr "script" ("type" =: "text/javascript") $ text $ mconcat $
+    [ "add_preload_tag = function (docSrc, docType) {       "
+    , "  var link_tag = document.createElement('link');     "
+    , "  link_tag.rel = 'preload';                          "
+    , "  link_tag.as = docType;                             "
+    , "  link_tag.href = docSrc;                            "
+    , "  document.head.appendChild(link_tag);               "
+    , "};                                                   "
+    , "if (typeof(SharedArrayBuffer) === 'undefined') {     "
+    , "  add_preload_tag('", allJsUrl, "', 'script');       "
+    , "} else {                                             "
+    , "  add_preload_tag('", jsaddleJs, "', 'script');      "
+    , "  add_preload_tag('", sendMsgWorkerJs, "', 'script');"
+    , "  add_preload_tag('", workerRunnerJs, "', 'script'); "
+    , "  add_preload_tag('", wasmUrl, "', 'script');        "
+    , "}"
+    ]
 
 -- | Load the script from the given URL in a deferred script tag.
 -- This is the default method.
 deferredGhcjsScript :: GhcjsWasmAssets -> FrontendWidgetT r ()
 deferredGhcjsScript (GhcjsWasmAssets allJsUrl mWasm) = case mWasm of
   Nothing -> elAttr "script" ("type" =: "text/javascript" <> "src" =: allJsUrl <> "defer" =: "defer") blank
-  Just v -> wasmScript v
+  Just wasmAssets -> wasmScript allJsUrl wasmAssets
 
-wasmScript :: (Text, Text) -> FrontendWidgetT r ()
-wasmScript (jsaddleAssets, wasmUrl) = do
+wasmScript :: Text -> (Text, Text) -> FrontendWidgetT r ()
+wasmScript allJsUrl (jsaddleAssets, wasmUrl) = do
   let sendMsgWorkerJs = jsaddleAssets <> "/jsaddle_sendMsgWorker.js"
       jsaddleJs = jsaddleAssets <> "/jsaddle.js"
       workerRunnerJs = jsaddleAssets <> "/worker_runner.js"
-  elAttr "script" ("type" =: "text/javascript" ) $ text $
-    "var jsaddle_sendMsgWorkerPath = '" <> sendMsgWorkerJs <> "';"
-  elAttr "script" ("type" =: "text/javascript" <> "src" =: jsaddleJs) blank
+      startScript = mconcat $
+        [ "var jsaddleVals = jsaddleJsInit();"
+        , "const worker = new Worker(\"" <> workerRunnerJs <> "\");"
+        , "worker.postMessage({ url: \"" <> wasmUrl <> "\""
+        , "                   , jsaddleVals: jsaddleVals }"
+        , "                   , [jsaddleVals.jsaddleListener]);"
+        ]
+
   elAttr "script" ("type" =: "text/javascript") $ text $ mconcat $
-    [ "var jsaddleVals = jsaddleJsInit();"
-    , "const worker = new Worker(\"" <> workerRunnerJs <> "\");"
-    , "worker.postMessage({ url: \"" <> wasmUrl <> "\""
-    , "                   , jsaddleVals: jsaddleVals }"
-    , "                   , [jsaddleVals.jsaddleListener]);"
+    [ "if (typeof(SharedArrayBuffer) === 'undefined') {           "
```

Comment:
You can use `Data.String.Here.Interpolated` to make this much less noisy.

Reply:
Either that or just use multiline strings since you don't need to manually include the `\n` in JS because it's not whitespace dependent.

Reply:
In fact, multiline strings would be an ideal way to make this readable but still achieve some form of "minimization" on the output.

Reply:
Done